tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -454,8 +406,9 @@ def _enable_disable_volume_profiling(self):
             cluster.volume_profiling_state = "mixed"
         volume_profiling_state = cluster.volume_profiling_state
         cluster = cluster.load()
-        cluster.volume_profiling_state = volume_profiling_state
-        cluster.save()
+        if cluster.volume_profiling_state != volume_profiling_state:
```

Comment:
Replace all   cluster.volume_profiling_state = "disabled", cluster.volume_profiling_state = "enabled", cluster.volume_profiling_state = "mixed" by volume_profiling_state = "disabled", volume_profiling_state = "enabled", volume_profiling_state = "mixed"

and modify the if condition like (if cluster.volume_profiling_state != volume_profiling_state and volume_profiling_state != None)

loading cluster object once is enough

Reply:
I agree to this. First calculate the value of new `volume_profiling_state` and at last load the cluster object and update the field.